### PR TITLE
Mark video as containing no segments

### PR DESCRIPTION
**Pull request by Joe-Dowd**
I've implemented updating the database with the categories you view, and comparing it with existing entries to add to the list. How that's used in submissions / how vips will submit is still to be discussed.

### CODE REVIEW

**Comment by Joe-Dowd** [comment]
Also includes bugfix for db schema updates.

File: src/routes/getIsUserVIP.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+var db = require('../databases/databases.js').db;
+
+var getHash = require('../utils/getHash.js');
+const logger = require('../utils/logger.js');
+const isUserVIP = require('../utils/isUserVIP.js');
+
+module.exports = function getUsername (req, res) {
```
Comment (ajayyy) [inline src/routes/getIsUserVIP.js:7]:
```suggestion
module.exports = (req, res) => {
```

File: src/routes/postNoSegments.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+const db = require('../databases/databases.js').db;
+const getHash = require('../utils/getHash.js');
+const isUserVIP = require('../utils/isUserVIP.js');
+const logger = require('../utils/logger.js');
+
+module.exports = (req, res) => {
+    // Collect user input data
+    let videoID = req.body.videoID;
+    let userID = req.body.userID;
+    let categorys = req.body.categorys;
```
Comment (ajayyy) [inline src/routes/postNoSegments.js:10]: `categorys` -> `categories`

File: src/routes/postNoSegments.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    // Check input data is valid
+    if (!videoID 
+        || !userID 
+        || !categorys 
+        || !Array.isArray(categorys) 
+        || categorys.length === 0
+    ) {
+        res.status(400).json({
+            status: 400,
```
Comment (ajayyy) [inline src/routes/postNoSegments.js:20]: I think it's redundant to include the status here. Just the message should be needed.

File: test/cases/getIsUserVIP.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
+var request = require('request');
+var utils = require('../utils.js');
+var db = require('../../src/databases/databases.js').db;
+var getHash = require('../../src/utils/getHash.js');
+
+describe('getSavedTimeForUser', () => {
```
Comment (ajayyy) [inline test/cases/getIsUserVIP.js:6]: Test name is wrong

File: src/routes/postNoSegments.js (hunk 3 above)
Comment (Joe-Dowd) [reply src/routes/postNoSegments.js:20]: as text or body.message?

Comment (Joe-Dowd) [reply src/routes/postNoSegments.js]: i went with body.message re https://github.com/ajayyy/SponsorBlockServer/issues/126

**Comment by Joe-Dowd** [comment]
chose category format check from 

```
sqlite>  select distinct category from sponsorTimes;
sponsor
selfpromo
outro
intro
music_offtopic
interaction
moreCategories
offtopic
```

File: src/app.js [hunk 5]
```diff
@@ -89,6 +91,13 @@ app.get('/api/getTotalStats', getTotalStats);
 //send out a formatted time saved total
 app.get('/api/getDaysSavedFormatted', getDaysSavedFormatted);
 
+//submit video containing no segments
+app.post('/api/postNoSegments', postNoSegments);
```
Comment (ajayyy) [inline src/app.js:95]:
```suggestion
app.post('/api/noSegments', postNoSegments);
```

File: src/app.js [hunk 6]
```diff
@@ -89,6 +91,13 @@ app.get('/api/getTotalStats', getTotalStats);
...
+
+//get if user is a vip
+app.get('/api/getIsUserVIP', getIsUserVIP);
```
Comment (ajayyy) [inline src/app.js:98]:
```suggestion
app.get('/api/isUserVIP', getIsUserVIP);
```

New format

**Review by ajayyy — APPROVED** [review]

**Pull request merged**